### PR TITLE
Support dependencies v2 in c*

### LINKED ISSUE #58: Update cassandra job to store dependencies into `dependencies_v2`
https://github.com/jaegertracing/jaeger/pull/1328/files#diff-4ac32a78649ca5bdd8e0ba38b7006a1eR15  introduced in Jaeger 1.10. Changed dependencies table from `dependencies` to `dependencies_v2`.

The c* job should be changed to write to this new table. Note that we do not version this library, we could do some magic e.g. find out of the table `dependencies_v2` exists and if yes store the result there.

@black-adder would you like to have a look at this one?

### PR DESCRIPTION
**Pull request by pavolloffay**
Resolves #58

### CODE REVIEW

**Comment by pavolloffay** [comment]
@black-adder could you please review this? I will merge it in order to fix CI in the operator.

File: jaeger-spark-dependencies-cassandra/src/main/java/io/jaegertracing/spark/dependencies/cassandra/CassandraDependenciesJob.java [hunk 1]
```diff
@@ -224,5 +245,32 @@ public Long getTsIndex() {
...
+
+    public List<Dependency> getDependencies() {
+      return dependencies;
+    }
+
+    public Long getTs() {
+      return zonedDateTime.toInstant().toEpochMilli();
+    }
+
+    public Long getTsBucket() {
```
Comment (pavolloffay) [inline jaeger-spark-dependencies-cassandra/src/main/java/io/jaegertracing/spark/dependencies/cassandra/CassandraDependenciesJob.java]: @black-adder is this ok?

File: jaeger-spark-dependencies-common/src/main/java/io/jaegertracing/spark/dependencies/model/Dependency.java [hunk 2]
```diff
@@ -66,6 +66,13 @@ public boolean equals(Object o) {
         && this.callCount == that.callCount;
   }
 
+  public String getSource() {
+    return "jaeger";
```
Comment (pavolloffay) [inline jaeger-spark-dependencies-common/src/main/java/io/jaegertracing/spark/dependencies/model/Dependency.java]: @black-adder is this ok?

Comment (black-adder) [reply jaeger-spark-dependencies-common/src/main/java/io/jaegertracing/spark/dependencies/model/Dependency.java]: yup

File: jaeger-spark-dependencies-common/src/main/java/io/jaegertracing/spark/dependencies/model/Dependency.java [hunk 3]
```diff
@@ -66,6 +66,13 @@ public boolean equals(Object o) {
...
+  }
+
+  public void setSource(String source) {
+  }
```
Comment (black-adder) [inline jaeger-spark-dependencies-common/src/main/java/io/jaegertracing/spark/dependencies/model/Dependency.java]: is this a no-op on purpose?

Comment (pavolloffay) [reply jaeger-spark-dependencies-common/src/main/java/io/jaegertracing/spark/dependencies/model/Dependency.java]: yep

File: jaeger-spark-dependencies-cassandra/src/main/java/io/jaegertracing/spark/dependencies/cassandra/CassandraDependenciesJob.java [hunk 4]
```diff
@@ -173,10 +174,19 @@ public void run() {
   }
 
   private void store(JavaSparkContext sc, List<Dependency> links) {
-    CassandraDependencies dependencies = new CassandraDependencies(links, day);
-    javaFunctions(sc.parallelize(Collections.singletonList(dependencies)))
-        .writerBuilder(keyspace, "dependencies", mapToRow(CassandraDependencies.class))
-        .saveToCassandra();
+    String table = dependenciesTable(sc);
+    log.info("Storing dependencies into {}", table);
+    if (table == "dependencies_v2") {
```
Comment (Sasasu) [inline jaeger-spark-dependencies-cassandra/src/main/java/io/jaegertracing/spark/dependencies/cassandra/CassandraDependenciesJob.java]: I think using equals() is better.
use == to compare strings, in this case, it works but makes confusing.

**Pull request merged**